athena-neptune/src/main/java/com/amazonaws/athena/connectors/neptune/rdf/RDFHandler.java

```diff
@@ -0,0 +1,180 @@
+     */
+
+    private final NeptuneSparqlConnection neptuneConnection;
+
+    // @VisibleForTesting
+    public RDFHandler(NeptuneConnection neptuneConnection) throws Exception
+    {
+        this.neptuneConnection = (NeptuneSparqlConnection) neptuneConnection;
+        this.neptuneConnection.connect();
+    }
+
+    /**
+     * Used to read the row data associated with the provided Split.
+     *
+     * @param spiller            A BlockSpiller that should be used to write the row
+     *                           data associated with this Split. The BlockSpiller
+     *                           automatically handles chunking the response,
+     *                           encrypting, and spilling to S3.
+     * @param recordsRequest     Details of the read request, including: 1. The
+     *                           Split 2. The Catalog, Database, and Table the read
+     *                           request is for. 3. The filtering predicate (if any)
+     *                           4. The columns required for projection.
+     * @param queryStatusChecker A QueryStatusChecker that you can use to stop doing
+     *                           work for a query that has already terminated
+     * @throws Exception
+     * @note Avoid writing >10 rows per-call to BlockSpiller.writeRow(...) because
+     *       this will limit the BlockSpiller's ability to control Block size. The
+     *       resulting increase in Block size may cause failures and reduced
+     *       performance.
+     */
+
+    public static final String PREFIX_KEY = "prefix_";
+    public static final int PREFIX_LEN = PREFIX_KEY.length();
+    
+
+    /**
+     * Performance considerations:
+     * If no constraints are provided, it gets THE WHOLE DATASET. That's so slow the Lamdbda this is running it might time out.
+     * Remedy:
+     * - Client always uses constraints to restrict output
+     * - For tables based on class, add a LIMIT that is configurable as lambda environment var
+     * - For tables based on sparql, allow LIMIT to be used
+     */
+    public void executeQuery(ReadRecordsRequest recordsRequest, final QueryStatusChecker queryStatusChecker,
+            final BlockSpiller spiller, java.util.Map<String, String> configOptions)  throws Exception 
+    {
+        // 1. Get the specified prefixes
+        Map<String, String> prefixMap = new HashMap<>();
+        String prefixBlock = "";
+        for (String k : recordsRequest.getSchema().getCustomMetadata().keySet()) {
+            if (k.startsWith(PREFIX_KEY)) {
+                String pfx = k.substring(PREFIX_LEN);
+                String val = recordsRequest.getSchema().getCustomMetadata().get(k);
+                prefixMap.put(pfx, val);
+                prefixBlock += "PREFIX " + pfx + ": <" + val + ">\n";
+            }
+        }
+
+        // 2. Build the SPARQL query
+        String queryMode = recordsRequest.getSchema().getCustomMetadata().get("querymode");
```

Comment:
pls use constants for all of these. Its hard to determine where they're coming from without prior context.

Reply:
changed